### ISSUE TITLE
Android: A lot of transactions can crash the App

### ISSUE THREAD

**Issue by ergovia-mobile** [litehelpers/Cordova-sqlcipher-adapter#3]
Hi,

when doing sort of heavy load on the database (50 INSERTS or so) we get following error and our app crashes. We are investigating at the moment. Does anyone have an idea? :)

````
03-27 14:41:23.680: E/System(13520): net.sqlcipher.database.SQLiteCompiledSql.finalize() timed out after 10000 ms; limit is 10000 ms
03-27 14:41:23.700: E/System(13520): java.util.concurrent.TimeoutException
03-27 14:41:23.700: E/System(13520): 	at java.lang.Object.wait(Native Method)
03-27 14:41:23.700: E/System(13520): 	at java.lang.Thread.parkFor(Thread.java:1231)
03-27 14:41:23.700: E/System(13520): 	at sun.misc.Unsafe.park(Unsafe.java:323)
03-27 14:41:23.700: E/System(13520): 	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:157)
03-27 14:41:23.700: E/System(13520): 	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:813)
03-27 14:41:23.700: E/System(13520): 	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireQueued(AbstractQueuedSynchronizer.java:846)
03-27 14:41:23.700: E/System(13520): 	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:1176)
03-27 14:41:23.700: E/System(13520): 	at java.util.concurrent.locks.ReentrantLock$FairSync.lock(ReentrantLock.java:200)
03-27 14:41:23.700: E/System(13520): 	at java.util.concurrent.locks.ReentrantLock.lock(ReentrantLock.java:261)
03-27 14:41:23.700: E/System(13520): 	at net.sqlcipher.database.SQLiteDatabase.lock(SQLiteDatabase.java:435)
03-27 14:41:23.700: E/System(13520): 	at net.sqlcipher.database.SQLiteCompiledSql.releaseSqlStatement(SQLiteCompiledSql.java:106)
03-27 14:41:23.700: E/System(13520): 	at net.sqlcipher.database.SQLiteCompiledSql.finalize(SQLiteCompiledSql.java:152)
03-27 14:41:23.700: E/System(13520): 	at java.lang.Daemons$FinalizerDaemon.doFinalize(Daemons.java:186)
03-27 14:41:23.700: E/System(13520): 	at java.lang.Daemons$FinalizerDaemon.run(Daemons.java:169)
03-27 14:41:23.700: E/System(13520): 	at java.lang.Thread.run(Thread.java:856)
````

**Comment by brodybits** [comment]
On Mar 27, 2015 2:46 PM, "ergovia-mobile" <notifications@github.com> wrote:
>
> Hi,
>
> when doing sort of heavy load on the database (50 INSERTS or so) we get
following error and our app crashes. We are investigating at the moment.
Does anyone have an idea? :)

I can take a look at this and #2 early next week. A small test program
would really help!

**Comment by ergovia-mobile** [comment]
Thx for the response. I will create a test program. Interestingly enough I cannot reproduce the error using the "old" Cordova-SQLitePlugin Tag r2014.07 with our extension for sqlcipher.

**Comment by ergovia-mobile** [comment]
Hi, any update on this? :)

**Comment by brodybits** [comment]
I can look at this in the next couple of weeks or so. Thanks for the reminder.

**Comment by johnnypark** [comment]
Same problem here, even only with a couple of Inserts (3 -5)

**Comment by rkistner** [comment]
While I could not consistently reproduce this issue, I did get it quite often. By closing the statements (see #15) I don't get this issue or the warnings in #2.

**Comment by brodybits** [comment]
Thanks @rkistner, I will check and merge PR #15 when I get a chance.

**Comment by ergovia-mobile** [comment]
I can confirm that my test case runs without an error. :+1:

**Comment by brodybits** [comment]
#15 is now merged, closing.

**Comment by brodybits** [comment]
From #15:
>> @brodybits it appears this PR had been accidentally reverted when you merged 'common-regexp-src' into cipher-rc ec5b50c
>> 
>> Any chance of this fix being re-applied?

> My apologies, I will reapply the fix when I get a chance. It will probably be a few weeks since I am busy with a couple of urgent projects right now. I will reopen #2 and #3 until I reapply the fix.

NOTE: This is marked as a duplicate since it is the *symptom* of #2.

**Comment by brodybits** [comment]
The fix is now applied.
